### PR TITLE
feat(modal): AUD-3785 add custom modal with scrolling content variant

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [ ] add custom `large` modal with independently scrollable columns

## Screenshots

Screenshot 2023-11-03 at 10 49 04 AM

## Testing in `sage-lib`

- Visit the [React Modal page](http://localhost:4100/?path=/docs/sage-modal--modal-with-scrolling-columns)
- Click on the Modal With Scrolling Columns example to view

**To be removed from Storybook before merge**

## Testing in `kajabi-products`

1. (**LOW**) Add custom modal variant that doesn't not exist in Storybook.

## Related

Closes [AUD-3785](https://kajabi.atlassian.net/browse/AUD-3785)

[AUD-3785]: https://kajabi.atlassian.net/browse/AUD-3785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pixelflips** [comment]
@QuintonJason I think the errors related to the dot.png are causing this one to not build properly. The following renders when I run it and I get a bunch of errors in the console.

Screenshot 2023-11-03 at 2 04 14 PM

File: packages/sage-assets/lib/stylesheets/components/_modal.scss [hunk 1]
```diff
@@ -347,3 +347,120 @@ $-modal-inner-size: sage-container(md);
...
+  border-inline-end: sage-border(default);
+  box-shadow: 6px 0 7px -6px rgba(sage-color(black), 0.16);
+
+  &.modal-scrollable-cols__fixed-column {
+    border-end-start-radius: sage-spacing(sm);
+  }
+}
+
+.modal-scrollable-cols__preview {
+  background-image: url("../../images/system/dot.png");
```
Comment (pixelflips) [inline packages/sage-assets/lib/stylesheets/components/_modal.scss:428]: @QuintonJason if the image keeps failing the build you might try base64 instead. That worked fine locally for me.

```
background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAALklEQVR4AdXPsQoAAASEYbz/syqLuomyi0n56rZ/OQYQtCC0dBCqeW0MO/zgdQLEyQ2Rj7XXawAAAABJRU5ErkJggg==');
```

Comment (QuintonJason) [reply packages/sage-assets/lib/stylesheets/components/_modal.scss]: Thanks for the suggestion

File: packages/sage-react/public/dot.png
Comment (pixelflips) [inline packages/sage-react/public/dot.png:1]: I think you can remove this file?

File: packages/sage-assets/lib/images/system/dot.png
Comment (pixelflips) [inline packages/sage-assets/lib/images/system/dot.png:1]: I think you can remove this file?

**Review by pixelflips — APPROVED** [review]
Really nice work here! LGTM 👍🏼 

A couple of non-blocking comments on deleting the image files and just as a note that the modal doesn't scroll on my screen unless I make the screen size small. Not an issue, just not sure if that's the intended behavior. 

<details>
  <summary>Screenshot</summary>
  Screenshot 2023-11-06 at 10 44 32 AM
</details>

**Review by anechol — APPROVED** [review]
LGTM

**Pull request merged**